lib/AuroraLip/Texture/Formats/GTX.cs

```diff
@@ -59,40 +74,87 @@ public struct GTXHeader
             public ushort Width;
             public ushort Height;
             public byte Bpp;
-            public byte Entries;
-            public ushort unk1; //0
+            /// <summary>
+            /// Base image + mipmaps, so minimum 1, maximum 8.
+            /// </summary>
+            public byte NumEntries;
+            /// <summary>
+            /// Mainly used for pool allocator, set to true while loading from file.
+            /// </summary>
+            public byte Used;
+            /// <summary>
+            /// Mainly set to 1 after GX calls, so overwritten during loading anyway.
+            /// </summary>
+            public byte InitedGX;
+            /// <summary>
+            /// Format of the texture.
+            /// </summary>
             public GTXFormat Format;
+            /// <summary>
+            /// Format of the palette. Should be filled out when the texture uses palette.
+            /// </summary>
             public GTXPaletteFormat PaletteFormat;
             //h10
             public uint WrapS;
             public uint WrapT;
-            public uint unk2; //0
-            public uint MagFilter;
+            public GTXFilterMode MinFilter;
```

Comment:
We could also use a full property with `get`.
```
            private uint magfilter;
```
```
            public GXFilterMode MagFilter
            {
                get => magfilter == 0x01 ? GXFilterMode.Linear : GXFilterMode.Nearest;
            }
```
this would save us the `If` checks directly in code and we could put the `set` code directly next to it.

Reply:
I am not sure about that. I interpreted that the struct describes how the format looks like and stuff like converting is handed off to the class that actually does the reading et cetera.

Is there any plans for writing textures back? Otherwise I don't think we need the setter.

Reply:
struct and class are very similar in c#, the main difference lies in the memory allocation.
You can leave it like this. This is just how I usually do it, because I find it more organized.

Yes, I have planned to add write functions, there is only so much that is more important.
which is why there is no point putting a lot of energy into it.